components/eventing-controller/controllers/backend/reconciler_internal_test.go

```diff
@@ -0,0 +1,94 @@
+//go:build unit
+// +build unit
+
+package backend
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/deployment"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TestGetSecretForPublisher verifies the successful and failing retrieval
+// of Secrets.
+func TestGetSecretForPublisher(t *testing.T) {
+	secretFor := func(message, namespace []byte) *corev1.Secret {
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: deployment.PublisherName,
+			},
+		}
+
+		secret.Data = make(map[string][]byte)
+
+		if len(message) > 0 {
+			secret.Data["messaging"] = message
+		}
+		if len(namespace) > 0 {
+			secret.Data["namespace"] = namespace
+		}
+
+		return secret
+	}
+
+	testCases := []struct {
+		name           string
+		messagingData  []byte
+		namespaceData  []byte
+		expectedSecret corev1.Secret
+		expectedError  error
+	}{
+		{
+			name: "with valid message and namespace data", // nolint:gofmt
+			messagingData: []byte("[{		\"broker\": {			\"type\": \"sapmgw\"		},		\"oa2\": {			\"clientid\": \"clientid\",			\"clientsecret\": \"clientsecret\",			\"granttype\": \"client_credentials\",			\"tokenendpoint\": \"https://token\"		},		\"protocol\": [\"amqp10ws\"],		\"uri\": \"wss://amqp\"	}, {		\"broker\": {			\"type\": \"sapmgw\"		},		\"oa2\": {			\"clientid\": \"clientid\",			\"clientsecret\": \"clientsecret\",			\"granttype\": \"client_credentials\",			\"tokenendpoint\": \"https://token\"		},		\"protocol\": [\"amqp10ws\"],		\"uri\": \"wss://amqp\"	},	{		\"broker\": {			\"type\": \"saprestmgw\"		},		\"oa2\": {			\"clientid\": \"rest-clientid\",			\"clientsecret\": \"rest-client-secret\",			\"granttype\": \"client_credentials\",			\"tokenendpoint\": \"https://rest-token\"		},		\"protocol\": [\"httprest\"],		\"uri\": \"https://rest-messaging\"	}]"),
```

Comment:
This is hard to read, please let's use a properly formatted string:

For example:

```golang
const s = `
{
   "key": "value"
}
`
```